### ISSUE TITLE
Create analytics-free flavor

### ISSUE THREAD

**Issue by loicortola** [jawg/osm-contributor#169]
To help with the deployment to alternative stores like F-Droid, all tier-libraries need to be FLOSS.
We should be fine within OSM Contributor, except for the Fabric.io analytics lib.

Suggestion:
Create a flavor without the library bundled into it.
This may require some refactoring and the creation of a wrapper with a "no-op" implementation for this flavor.

This issue is related to discussions ongoing in issue #166

**Comment by rugk** [comment]
Remember that you must not compile the properitary lib into the app, when you want to get on F-Droid. Just not using/disabling it with a wrapper is not enough.

**Comment by loicortola** [comment]
Absolutely, although you need a wrapper interface with two implementations: one that does no-op, one that calls the library.
If we don't do this, removing the library will not compile ;)

**Comment by IzzySoft** [comment]
Any news on this, @loicortola? The [issue at F-Droid](https://gitlab.com/fdroid/rfp/issues/358) is still stalled waiting on this.

**Comment by rugk** [comment]
...or, alternatively, you switch to a FLOSS analytics library. (you get an antifeature badge then, but at least you are in F-Droid)

**Comment by IzzySoft** [comment]
As @rugk says. Two examples would be ACRA and Matomo, so you have some names at least.

**Comment by tuxayo** [comment]
In the hope to move this forward a bounty has been started:

[![Bountysource](
https://api.bountysource.com/badge/issue?issue_id=61636970
)](
https://www.bountysource.com/issues/61636970-create-analytics-free-flavor
)

**Comment by hferee** [comment]
I've started working on this, until I realised that Fabric is deprecated and, if I understand correctly, will be unusable in about a month from now. 
Am I correct that that it is not worth making this no-op class anymore?
Should we instead focus on what analytics library you want to switch to?

**Comment by IzzySoft** [comment]
Mentioning another acceptable analytics framework from [my F-Droid snippet](https://gitlab.com/snippets/1896503#acceptable-analytics): [tracedroid](https://github.com/ligi/tracedroid) ([ACRA](https://github.com/ACRA/acra) I've already mentioned above). If you stumble upon one I have not yet listed in said snippet, I gladly take hints!

**Comment by loicortola** [comment]
Hello guys,

latest update for JungleBus has been made last month and upgraded Fabric to Firebase.
We would be very interested if someone wanted to submit a PR with a tracedroid-ready flavor.
